Okay, libdbus used to generate a warning if we used dbus_connection_close(),
but now it aborts.

diff --git a/src/oddjob_dbus.c b/src/oddjob_dbus.c
index f26bd63..6660007 100644
--- a/src/oddjob_dbus.c
+++ b/src/oddjob_dbus.c
@@ -1,5 +1,5 @@
 /*
-   Copyright 2005,2006,2007 Red Hat, Inc.
+   Copyright 2005,2006,2007,2010 Red Hat, Inc.
    All rights reserved.
 
    Redistribution and use in source and binary forms, with or without
@@ -138,7 +138,8 @@ void
 oddjob_dbus_connection_close(DBusConnection *conn)
 {
 #if DBUS_CHECK_VERSION(0,34,0)
-        dbus_connection_close(conn);
+	/* Apparently we abort now when we try this. */
+        /* dbus_connection_close(conn); */
 #elif DBUS_CHECK_VERSION(0,20,0)
 	dbus_connection_disconnect(conn);
 #else
@@ -237,8 +238,7 @@ oddjob_dbus_listener_free(struct oddjob_dbus_context *ctx)
 					      ctx);
 		ctx->registered = FALSE;
 	}
-	/* Apparently we abort now when we try this. */
-	/* oddjob_dbus_connection_close(ctx->conn); */
+	oddjob_dbus_connection_close(ctx->conn);
 	ctx->conn = NULL;
 	oddjob_free(ctx);
 }
